app/views/partials/yourSelfAssessmentCharges/PaymentHistoryTable.scala.html

```diff
@@ -0,0 +1,173 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *@
+
+@import uk.gov.hmrc.govukfrontend.views.html.components._
+@import _root_.implicits.ImplicitCurrencyFormatter.CurrencyFormatter
+@import _root_.implicits.ImplicitDateFormatterImpl
+@import _root_.implicits.HtmlFormatter.NbspString
+@import views.html.components.h2
+@import models.chargeSummary.ChargeSummaryViewModel
+@import views.html.partials.yourSelfAssessmentCharges.LatePaymentInterestChargeMessage
+@import models.repaymentHistory.RepaymentHistoryUtils
+@import exceptions.MissingFieldException
+@import java.time.LocalDate
+@import models.financialDetails.PoaOneDebit
+@import models.financialDetails.PoaTwoDebit
+
+@this(
+        govukTable: GovukTable,
+        implicitDateFormatter: ImplicitDateFormatterImpl,
+        latePaymentInterestChargeMessage: LatePaymentInterestChargeMessage,
+        h2: h2
+)
+
+@(viewModel: ChargeSummaryViewModel)(implicit messages: Messages)
+
+@import implicitDateFormatter.longDate
+
+@reviewAndReconcileCreditMessage = {
+    <a class="govuk-link" id="rar-charge-link" href="@RepaymentHistoryUtils.getPoaChargeLinkUrl(viewModel.isAgent, viewModel.chargeItem.taxYear.endYear, viewModel.chargeItem.transactionId)">
+        @messages(s"chargeSummary.chargeHistory.${viewModel.chargeItem.transactionType.key}")
+    </a>
+}
+
+@getCreationDate(date: Option[LocalDate]) = @{
+    date match {
+        case Some(value) => value.toLongDate
+        case None => messages("incomeSources.generic.unknown")
+    }
+}
+
+@chargeHistoryHeading = @{
+    viewModel.chargeItem.transactionType match {
+        case _ if viewModel.latePaymentInterestCharge => messages("chargeSummary.chargeHistory.lateInterestPayment")
+        case PoaOneDebit                              => messages("chargeSummary.chargeHistory.Poa1heading")
+        case PoaTwoDebit                              => messages("chargeSummary.chargeHistory.Poa2heading")
+        case _                                        => messages("chargeSummary.chargeHistory.heading")
+    }
+}
+
+@govukTable(Table(
+    head = Some(
+        Seq(
+            HeadCell(
+                content = Text(messages("chargeSummary.chargeHistory.description"))
+            ),
+            HeadCell(
+                content = HtmlContent(messages("chargeSummary.chargeHistory.date"))
+            ),
+            HeadCell(
+                content = Text(messages("chargeSummary.chargeHistory.amount")),
+                format = Some("numeric")
+            )
+        )
+    ),
+    rows = Seq(
+        if(viewModel.noInterestChargeAndNoCodingOutEnabledWithIsPayeSelfAssessment)
```

Comment:
might worth to refactor this large segment of code where we joining results with ++;
or I guess to create a tech debt story if we are in the rush to push this part of the work

Reply:
we can do this before passing view model into the scope of the view ie; or say create a new property in the view model etc; the main reason to mention this was that we have to reduce this complexity in order to make future maintenance much easier + we have to consider that we would need to extend this view level logic if requirements would change; might worth to create a tech debt ? really up to you